lib/src/event_controller.dart

```diff
@@ -181,6 +201,33 @@ class EventController<T extends Object?> extends ChangeNotifier {
     notifyListeners();
   }
 
+  /// Selected eventData in  [DayView], this can be expanded to include other 
+  /// views.
+  /// 
+  /// This is so that the event can be highlighted and modified by the user.
+  CalendarEventData<T>? selectedEventData;
```

Comment:
I'm not sure if this direction is acceptable if it isn't can we please discuss how to implement this feature correctly.